RadixWallet/Features/GatewaySettingsFeature/Children/AddNewGateway/AddNewGateway+Reducer.swift

```diff
@@ -102,7 +96,11 @@ public struct AddNewGateway: Sendable, FeatureReducer {
 			return handle(error, state: &state)
 
 		case .addGatewayResult(.success):
-			return .send(.delegate(.dismiss))
+			return .run { _ in
+				if isPresented {
```

Comment:
do we need the `isPresented` check?

Reply:
Without this check, the `test_whenNewAddGatewayButtonIsTapped_thenDelegateIsCalled` test fails:
```
A reducer requested dismissal at "Radix_Wallet_Dev/AddNewGateway+Reducer.swift:110", but couldn't be dismissed. …

This is generally considered an application logic error, and can happen when a reducer assumes it runs in a presentation context. If a reducer can run at both the root level of an application, as well as in a presentation destination, use @Dependency(\.isPresented) to determine if the reducer is being presented before calling @Dependency(\.dismiss).
```